doc/howtos/rdtraining/inheritance.rst

```diff
@@ -0,0 +1,241 @@
+.. _howto/rdtraining/inheritance:
+
+===========
+Inheritance
+===========
+
+A powerful aspect of Odoo is its modularity. A module is dedicated to a business need, but moreover
+modules can interact one with another. This is useful to extend the functionality of an existing
+module. For example, in out real estate scenario we want to display the list of properties
+of a salesman, directly on the regular user view. 
+
+But before going through the specific Odoo module inheritance, let's see how we can alter the
+behavior of the regular CRUD methods. 
+
+Python Inheritance
+==================
+
+.. note::
+
+    **Goal**: at the end of this section:
+
+    - It is not possible to delete a property which is not new or sold.
+
+    .. image:: inheritance/media/unlink.gif
+        :align: center
+        :alt: Unlink
+
+    - When an offer is created, the property state changes to 'Offer Received'
+    - It is not possible to create an offer with a lower price than an existing offer
+
+    .. image:: inheritance/media/create.gif
+        :align: center
+        :alt: Create
+
+In our real estate module, we never had to develop anything specific feature to be able to do any
+regular CRUD (Create, Retrieve, Update or Delete) action. The Odoo framework provides the necessary
+tools to do it. Indeed, such actions are already included in our model thanks to the classical
+Python inheritance::
+
+    from odoo import fields, models
+
+    class TestModel(models.Model):
+        _name = "test.model"
+        _description = "Test Model"
+
+        ...
+
+Our ``class TestModel`` inherits from the :class:`~odoo.models.Model` which provides the
+:meth:`~odoo.models.Model.create`, :meth:`~odoo.models.Model.read`, :meth:`~odoo.models.Model.write`
+and :meth:`~odoo.models.Model.unlink`.
+
+These methods (and any other method defined on :class:`~odoo.models.Model`) can be extended to add
+specific business logic::
+
+    from odoo import fields, models
+
+    class TestModel(models.Model):
+        _name = "test.model"
+        _description = "Test Model"
+
+        ...
+
+        @api.model
+        def create(self, vals):
+        # Do some business logic, modify vals...
+        ...
+        # Then call super to execute the parent method 
+        return super().create(vals)
+
+The decorator :func:`~odoo.api.model` is only necessary for the :meth:`~odoo.models.Model.create`
+method since the content of the recordset ``self`` is not relevant in the context of a creation.
+
+.. danger::
+
+    - It is very important to **always** call ``super()`` to avoid breaking the flow. There are
```

Comment:
Maybe explain that they're going to find in the sources something like : super(Class_Name, self), that is equivalent to super().